src/arrayMethodSort.js

```diff
@@ -4,9 +4,42 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = defaultCompareFunction) {
+    let hadChanges;
+
+    do {
+      hadChanges = false;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        if (compareFunction(this[i], this[i + 1]) > 0) {
+          switchPlaces(this, i, i + 1);
+          hadChanges = true;
+        }
+      }
+    } while (hadChanges);
+
+    return this;
   };
 }
 
+function switchPlaces(arr, index1, index2) {
+  const memory = arr[index1];
+
+  arr[index1] = arr[index2];
+  arr[index2] = memory;
+}
+
+function defaultCompareFunction(a, b) {
+  const first = '' + a;
+  const second = '' + b;
+
+  switch (true) {
+    case first < second: return -1;
+    case first > second: return 1;
+    case first === second: return 0;
+    default:
+      throw new Error('some error'); // I have no idea what is it.
+  }
```

Comment:
In the context of this function return 0 will be your default case, because you don't need to sort elements if they are the same. So you don't need that error here
```suggestion
    case first === second:
    default:
    	return 0;
  }
```

or it can work with just `default: return 0`